hack/tools/catalogs/download-catalog

```diff
@@ -0,0 +1,72 @@
+#!/usr/bin/env bash
+
+SCRIPT_ROOT=$(dirname "$(realpath "$0")")
+source "${SCRIPT_ROOT}/lib/utils.sh"
+
+# Constants
+SERVICE_NAMESPACE="olmv1-system"
+SERVICE_NAME="catalogd-catalogserver"
+LOCAL_PORT=8001
+SERVICE_PORT=443  # Assuming the service uses HTTPS on port 443
+
+# Display usage
+usage() {
+    print-banner
+    echo ""
+    echo "Usage: $0 <catalog-name>"
+    echo ""
+    echo "Download catalog from a ClusterCatalog in a cluster reachable from KUBECONFIG"
+    echo "Downloaded catalog will be saved as <catalog-name>-catalog.json"
+    echo ""
+    echo "Example:"
+    echo "  $0 operatorhubio"
+}
+
+# Check if catalog name is provided
+if [ -z "$1" ]; then
+    usage
+    exit 1
+fi
+
+CATALOG_NAME="$1"
+
+# Check if the clustercatalog resource exists
+echo "Checking if ClusterCatalog $CATALOG_NAME exists..."
+CLUSTER_CATALOG=$(kubectl get clustercatalog "$CATALOG_NAME" -o json 2>/dev/null)
```

Comment:
Should we have a check/documentation that lets a user know that `kubectl` is required?

Reply:
I have added a README to the directory and also assertions at the beginning of each script.